x-pack/plugins/lens/public/pie_visualization/toolbar.tsx

```diff
@@ -131,59 +183,26 @@ export function PieToolbar(props: VisualizationToolbarProps<PieVisualizationStat
         >
           <DecimalPlaceSlider
             value={layer.percentDecimals ?? DEFAULT_PERCENT_DECIMALS}
-            setValue={(value) => {
-              setState({
-                ...state,
-                layers: [{ ...layer, percentDecimals: value }],
-              });
-            }}
+            setValue={onPercentDecimalsChange}
           />
         </EuiFormRow>
       </ToolbarPopover>
       <LegendSettingsPopover
         legendOptions={legendOptions}
         mode={layer.legendDisplay}
-        onDisplayChange={(optionId) => {
-          setState({
-            ...state,
-            layers: [
-              {
-                ...layer,
-                legendDisplay: legendOptions.find(({ id }) => id === optionId)!.value,
-              },
-            ],
-          });
-        }}
+        onDisplayChange={onLegendDisplayChange}
+        valueInLegend={shouldShowValuesInLegend(layer, state.shape)}
+        renderValueInLegendSwitch={Boolean(PartitionChartsMeta[state.shape]?.legend?.showValues)}
```

Comment:
@flash1293 @ghudgins added "**Show Label**" for **Waffle** only. What are your thoughts on adding this property for other partition charts?

Reply:
I can't see any downsides to do this. on the mosaic specifically i'm not sure it makes sense to display overall metrics in the legends given you aren't visualizing that metric but maybe it's a nice piece of extra information when you aren't hovering?

Reply:
+1 for just allowing this for waffle for now.